src/include/detail/graph/vamana.h

```diff
@@ -222,7 +222,7 @@ auto greedy_search(
         // @todo (actually it does, but shouldn't need it -- need to
         // investigate) if (result.template insert /*<unique_id>*/ (score, p)) {
         if (result.template insert<unique_id>(score, p)) {
-          q2.template insert /*<unique_id>*/ (score, p);
+          q2.template insert<unique_id>(score, p);
```

Comment:
Is this change required? The comment above seems to indicate it makes a difference in something? cc @lums658 if you could explain the comment a bit more?

Reply:
Ah, yes. This was a compilation issue for MSVC, saying that it did not expect word `template`.

Reply:
Oh this is one of the issues that has been impacting performance -- whether or not to explicitly maintain unique keys in the heap.  Better in this case to take `template` out rather than put `unique_id` back in: 
```c++
         if (result.template insert<unique_id>(score, p)) {
           // q2.template insert <unique_id>(score, p);
           q2.insert(score, p);
```